test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/RequestTests.cs

```diff
@@ -566,6 +567,83 @@ public void CanUpgradeRequestWithConnectionKeepAliveUpgradeHeader()
             }
         }
 
+        [Fact]
+        public async Task AppCanSetTraceIdentifier()
+        {
+            const string knownId = "xyz123";
+            using (var server = new TestServer(async context =>
+            {
+                context.TraceIdentifier = knownId;
+                await context.Response.WriteAsync(context.TraceIdentifier);
+            }))
+            {
+                var requestId = await HttpClientSlim.GetStringAsync($"http://{server.EndPoint}")
+                    .TimeoutAfter(TimeSpan.FromSeconds(10));
+                Assert.Equal(knownId, requestId);
+            }
+        }
+
+        [Fact]
+        public async Task TraceIdentifierIsUnique()
+        {
+            const int IdentifierLength = 13;
+            const int iterations = 100;
+
+            using (var server = new TestServer(context =>
+            {
+                var id = Encoding.ASCII.GetBytes(context.TraceIdentifier);
+                Assert.Equal(IdentifierLength, id.Length);
+                context.Response.ContentLength = id.Length;
+                context.Response.Body.Write(id, 0, id.Length);
```

Comment:
Returns chunked encoding, which made it more difficult to pull the response body via StreamReader.